rules/catch-potential-xss-vue.js

```diff
@@ -15,78 +15,315 @@ const isLibraryTrusted = (source) => {
-      return isMemberExpressionSafe(node, isVariableTrusted);
+const checkNode = (currentNode, isVariableTrusted, variableNameToBeAssigned = '') => {
+  if (!currentNode) {
+    return;
+  }
+  switch (currentNode.type) {
+    case 'Program' :
+      for(const nextNode of currentNode.body) {
+        checkNode(nextNode, isVariableTrusted)
+      }
+      break;
+    // Declarations
+    case 'ImportDeclaration':
+      const { specifiers, source } = currentNode;
+      for (const specifier of specifiers) {
+        const name = get(specifier, 'local.name', '');
+        if (name !== '') {
+          updateIsvariableTrusted(isVariableTrusted, name, {value: isLibraryTrusted(source.value), dependsOn: []});
+        }
+      }
+      break;
+    case 'ExportDefaultDeclaration':
+      checkNode(currentNode.declaration, isVariableTrusted)
+      break;
+    case 'VariableDeclaration':
+      for(const nextNode of currentNode.declarations) {
+        checkNode(nextNode, isVariableTrusted)
+      }
+      break;
+    case 'VariableDeclarator':
+      updateIsvariableTrusted(isVariableTrusted, currentNode.id.name, checkNode(currentNode.init, isVariableTrusted, currentNode.id.name))
+      break;
+    case 'FunctionDeclaration':
+      checkNode(currentNode.body, isVariableTrusted);
+      updateIsvariableTrusted(isVariableTrusted, currentNode.id.name, checkReturnsInStatement(currentNode.body, isVariableTrusted))
+      break;
+    // Statements and Clauses
+    case 'BlockStatement':
+      for(const nextNode of currentNode.body) {
+        checkNode(nextNode, isVariableTrusted)
+      }
+      break;
+    case 'IfStatement':
+      checkNode(currentNode.consequent, isVariableTrusted);
+      checkNode(currentNode.alternate, isVariableTrusted);
+      break;
+    case 'SwitchStatement':
+      for(const switchCase of currentNode.cases) {
+        checkNode(switchCase, isVariableTrusted)
+      }
+      break;
+    case 'SwitchCase':
+      for(const consequent of currentNode.consequent) {
+        areReturnsInStatementSafe(consequent, isVariableTrusted)
+      }
+      break;
+    case 'TryStatement':
+      checkNode(currentNode.block, isVariableTrusted);
+      for(const catchClause of currentNode.handler) {
+        checkNode(catchClause, isVariableTrusted);
+      }
+      break;
+    case 'CatchClause':
+      checkNode(currentNode.body, isVariableTrusted);
+      break;
+    case 'WhileStatement':
+    case 'DoWhileStatement':
+    case 'ForStatement':
+    case 'ForInStatement':
+    case 'ForOfStatement':
+      checkNode(currentNode.body, isVariableTrusted)
+      break;
+    case 'ExpressionStatement':
+      return checkNode(currentNode.expression, isVariableTrusted, variableNameToBeAssigned);
+    // Expressions
+    case 'AssignmentExpression':
+      const variableName = getNameFromExpression(currentNode.left);
+      updateIsvariableTrusted(isVariableTrusted, variableName, checkNode(currentNode.right, isVariableTrusted, variableName));
```

Comment:
CamelCase sur updateIsVariableTrusted :)

Reply:
Done it in the next ticket (while deplacing it to react). Sorry it's not in this PR :/